llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -11660,6 +11660,24 @@ SDValue DAGCombiner::visitSELECT(SDNode *N) {
     }
   }
 
+  // select m, sub(X, C), X --> sub (X, and(C, m))
```

Comment:
I'm confused. Shouldn't select have a scalar condition. If the condition is a vector it should be ISD::VSELECT

Reply:
These folds need moving to visitVSELECT